cmd/sql/flow.go

```diff
@@ -4,273 +4,235 @@ import (
+		}
+		paths = append(paths, readConfigCmdOutput("airflow_home"))
+	case generateCmdName:
+		if projectDir != "" {
+			paths = append(paths, projectDir)
+		}
+		paths = append(paths, readConfigCmdOutput("airflow_home"), readConfigCmdOutput("airflow_dags_folder"))
+	case runCmdName:
+		if projectDir != "" {
+			paths = append(paths, projectDir)
+		}
+		paths = append(paths, readConfigCmdOutput("airflow_home"), readConfigCmdOutput("airflow_dags_folder"))
 	}
-
-	if setAirflowDagsFolder && airflowDagsFolder != "" {
-		airflowDagsFolderAbs, err := getAbsolutePath(airflowDagsFolder)
-		if err != nil {
-			return nil, nil, err
+	mounts := make([]string, len(paths))
+	for i, path := range paths {
+		absPath := resolvePath(path)
+		if err := os.MkdirAll(absPath, os.ModePerm); err != nil {
+			panic(err)
 		}
-		flags["airflow-dags-folder"] = airflowDagsFolderAbs
-		mountDirs = append(mountDirs, airflowDagsFolderAbs)
+		mounts[i] = absPath
 	}
-
-	return flags, mountDirs, nil
+	return mounts
 }
 
-func executeCmd(cmd *cobra.Command, args []string, flags map[string]string, mountDirs []string) error {
-	cmdString := []string{cmd.Name()}
-	if debug {
-		cmdString = []string{"--debug", cmd.Name()}
-	}
-	exitCode, _, err := sql.ExecuteCmdInDocker(cmdString, args, flags, mountDirs, false)
+// Execute cobra cmd with args
+func executeCmd(cmd *cobra.Command, args []string) error {
+	cmdString := buildCmd(cmd, args)
+	exitCode, _, err := sql.ExecuteCmdInDocker(cmdString, createDirs(cmd), false)
 	if err != nil {
 		return fmt.Errorf("error running %v: %w", cmdString, err)
 	}
 	if exitCode != 0 {
 		return sql.DockerNonZeroExitCodeError(exitCode)
 	}
-
 	return nil
 }
 
-func executeBase(cmd *cobra.Command, args []string) error {
-	flags, mountDirs, err := buildFlagsAndMountDirs(projectDir, false, false, false, false)
-	if err != nil {
-		return err
-	}
-	return executeCmd(cmd, args, flags, mountDirs)
-}
-
-func executeInit(cmd *cobra.Command, args []string) error {
-	if len(args) > 0 {
-		projectDir = args[0]
-	}
-
-	flags, mountDirs, err := buildFlagsAndMountDirs(projectDir, false, true, true, false)
-	if err != nil {
-		return err
-	}
-
-	projectDirAbsolute := mountDirs[0]
-	args = []string{projectDirAbsolute}
-
-	return executeCmd(cmd, args, flags, mountDirs)
-}
-
-func executeConfig(cmd *cobra.Command, args []string) error {
-	if len(args) < 1 {
-		return sql.ArgNotSetError("key")
-	}
-
-	flags, mountDirs, err := buildFlagsAndMountDirs(projectDir, true, false, false, false)
-	if err != nil {
-		return err
-	}
-
-	if environment != "" {
-		flags["env"] = environment
-	}
-
-	return executeCmd(cmd, args, flags, mountDirs)
-}
-
-func executeValidate(cmd *cobra.Command, args []string) error {
-	if len(args) > 0 {
-		projectDir = args[0]
-	}
-
-	flags, mountDirs, err := buildFlagsAndMountDirs(projectDir, false, false, false, false)
+// Read cmd output from cobra cmd with args
+func readCmdOutput(cmd *cobra.Command, args []string) string {
```

Comment:
should we rename this to `executeAndReadCmdOutput`?

Reply:
I used python's [subprocess.check_output](https://docs.python.org/3/library/subprocess.html#subprocess.check_output) as a guide. The execution in that case is implicit. I can improve the docstring WDYT?

Reply:
I have updated the docstring.